File: leetcode/001-Two-Sum/csharp/Solution.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> numIdx = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++) {
            if (numIdx.ContainsKey(target - nums[i])) {
                return new int[]{numIdx[target - nums[i]], i};
            }
            numIdx[nums[i]] = i;
        }
        return new int[]{};
    }
}

File: leetcode/001-Two-Sum/rust/Solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {

        let mut numIdx: HashMap<i32, i32> = HashMap::new();
        for (i, num) in nums.iter().enumerate() {
            if numIdx.contains_key(&(target - num)) {
                return vec![numIdx[&(target - num)], i as i32];
            }
            numIdx.insert(*num, i as i32);
        }
        return vec![-1, -1];
    }
}